### PR TITLE
Fixes #25546 - assert action state in HypervisorsUpdateTest

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

**Comment by evgeni** [comment]
This intentionally breaks tests to expose the current issue.

@ShimShtein @jlsherrill @thomasmckay, you were the last ones to touch these tests and I'd appreciate your thoughts on this.

**Comment by evgeni** [comment]
Ok, now the tests pass, but I am not 100% sure what I did is valid, so please have a good :eyes: here

File: app/lib/actions/katello/host/hypervisors_update.rb [hunk 1]
```diff
@@ -40,7 +40,10 @@ def load_resources
 
         def load_hosts_by_uuid
           hosts_by_uuid = ::Host.eager_load(:subscription_facet).where(katello_subscription_facets: { uuid: hypervisors_field(:uuid) })
-          @hosts.merge(hosts_by_uuid.map { |host| [host.subscription_facet.uuid, host] }.to_h)
```
Comment (beav) [inline app/lib/actions/katello/host/hypervisors_update.rb]: was this a no-op previously?

Comment (evgeni) [reply app/lib/actions/katello/host/hypervisors_update.rb]: It didn't update @hosts, so pretty much yeah

**Comment by beav** [comment]
@evgeni the change looks good, but can you break the `@hosts.merge!` up one level into `load_resources`, just so it's not as hidden?

for example, something like

```ruby
@hosts.merge!(load_hosts_by_uuid)
@hosts.merge!(load_hosts_by_duplicate_name)
@hosts.merge!(create_missing_hosts)
```

**Comment by evgeni** [comment]
@beav done

**Comment by beav** [comment]
@evgeni I hit approve but then thought of a question.

Just confirming to make sure I read it correctly, with the change, virt-who will now error if any uuids are reported that are for machines in a different org? I think this can happen sometimes, so I don't know if it's OK or not.

**Comment by evgeni** [comment]
Well, yeah, but it also expected that error before, just didn't really receive it

**Comment by evgeni** [comment]
virt who itself will not error on this, the update task that happens after the check in might

**Comment by evgeni** [comment]
and now on a proper keyboard, not some mobile phone:

This PR changes two things:
* makes `load_hosts_by_uuid` actually work
* makes `load_hosts_by_uuid` verify the org of the host

The hosts that weren't found by UUID before, were found by name in `load_hosts_by_duplicate_name` and that also did verification of the org, so I don't think this PR actually changes our error behavior.

Unless we report the same UUID in two orgs? Is that what you were referring to @beav?

**Comment by beav** [comment]
@evgeni ah I think I understand now. So the check is only checking that the host has any org associated with it, correct?

**Comment by beav** [comment]
ack thanks @evgeni !

**Pull request merged**